helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -552,7 +552,7 @@ private static PipelineRegistry createDefaultRegistry(String pipelineName) {
       autoExitMaintenancePipeline.addStage(new MaintenanceRecoveryStage());
 
       registry.register(ClusterEventType.IdealStateChange, dataRefresh, dataPreprocess,
-          rebalancePipeline);
+          externalViewPipeline, rebalancePipeline);
```

Comment:
It is non-deterministic test which we all agree as test is not failing consistently in CI. Just curious, changing the code is the right behavior or should mock<> implementation of controller needs to change??

Reply:
Right now I doubt the non-deterministic part, like, what if we have `setupTool.dropResourceFromCluster` long enough after `participant.syncStop()`, will the test still pass occasionally? I documented something in description regarding ExternalViewComputeStage.
This is a good topic for discussion, we need to review and define the expected behavior. If the test is describing the right behavior, then it's a bug in code we need to fix.